File: TFE.Api/Models/Notification.cs
namespace TFE.Api.Models;

public class Notification
{
    public Guid Id { get; set; }

    public string UserId { get; set; } = string.Empty;
    public virtual ApplicationUser User { get; set; } = null!;

    public Guid PatientId { get; set; }
    public virtual Patient Patient { get; set; } = null!;

    public string Title { get; set; } = string.Empty;
    public string Message { get; set; } = string.Empty;
    public bool IsRead { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    /// <summary>
    /// Client-side deep-link, e.g. /patients/{id}?postId={postId}&amp;commentId={commentId}.
    /// Null for legacy notifications that pre-date this field.
    /// </summary>
    public string? TargetUrl { get; set; }
}

File: frontend/src/types/notification.ts
export type NotificationType = 'NewPost' | 'NewComment' | 'PatientUpdated';

export interface NotificationResponse {
  id: string;
  patientId: string;
  type: NotificationType;
  actorFirstName: string;
  actorLastName: string;
  patientFirstName: string;
  patientLastName: string;
  isRead: boolean;
  createdAt: string;
  /** Deep-link to the targeted content, e.g. /patients/{id}?postId={postId}&commentId={commentId}. */
  targetUrl?: string;
}
